File: Main/Models/Element.cs

using Newtonsoft.Json;

namespace LinkCollectionApp.Models
{
  public partial class Element
  {
    public int Id { get; set; }
    public int CollectionId { get; set; }
    public string OwnerId { get; set; }
    public string Link { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public int? Sequence { get; set; }

    [JsonIgnore]
    public virtual Collection Collection { get; set; }
    [JsonIgnore]
    public virtual ApplicationUser Owner { get; set; }
  }
}

File: Main/ClientApp/src/Model/Element.ts
export type Element = {
  id: number;
  collectionId: number;
  ownerId: string;
  link: string;
  name: string;
  description: string;
  sequence: number | null;
};

export type ElementCreationData = {
  collectionId: number;
  link: string;
  name: string;
};

export type ElementUpdateData = {
  link: string;
  name: string;
};
